### PR TITLE
Use semver for version comparison

### LINKED ISSUE #2644: Use semver for version comparison
https://github.com/composer/semver

### PR DESCRIPTION
**Pull request by nitriques**
Fixes #2644

### CODE REVIEW

File: install/lib/class.installer.php [hunk 1]
```diff
@@ -194,7 +194,7 @@ private function checkRequirements()
         $errors = [];
 
         // Check for PHP 5.6+
-        if (version_compare(phpversion(), '5.6', '<=')) {
+        if (\Composer\Semver\Comparator::lessThan(phpversion(), '5.6')) {
             $errors[] = array(
                 'msg' => __('PHP Version is not correct'),
                 'details' => __('Symphony requires %1$s or greater to work, however version %2$s was detected.', array('<code><abbr title="PHP: Hypertext Pre-processor">PHP</abbr> 5.3</code>', '<code>' . phpversion() . '</code>'))
```
Comment (fhamon) [inline install/lib/class.installer.php]: The text will say that you need at least php 5.3.

File: install/lib/class.installer.php [hunk 2]
```diff
@@ -318,7 +318,7 @@ private function checkConfiguration()
             // Check the database credentials
             elseif ($db && $db->isConnected()) {
                 // Incorrect MySQL version
-                if (version_compare($db->getVersion(), '5.5', '<')) {
+                if (\Composer\Semver\Comparator::lessThan($db->getVersion(), '5.6')) {
                     $errors['database-incorrect-version']  = array(
                         'msg' => 'MySQL Version is not correct. '. $version . ' detected.',
                         'details' => __('Symphony requires %1$s or greater to work, however version %2$s was detected. This requirement must be met before installation can proceed.', array('<code>MySQL 5.5</code>', '<code>' . $version . '</code>'))
```
Comment (fhamon) [inline install/lib/class.installer.php]: The text will say that you need at least MySQL 5.5

File: install/lib/class.updater.php [hunk 3]
```diff
@@ -127,7 +130,7 @@ public function run()
             );
 
             $this->render(new UpdaterPage('uptodate'));
-        } elseif (version_compare($currentSymphony, '2.7.0', '<')) {
+        } elseif (\Composer\Semver\Comparator::lessThan($currentSymphony, '2.7.0')) {
```
Comment (fhamon) [inline install/lib/class.updater.php]: Should 2.7.0 be replaced by 3.0.0 ? @nitriques not sure here

File: install/lib/class.installer.php (hunk 1 above)
Comment (michael-e) [reply install/lib/class.installer.php]: Good catch!

File: install/lib/class.installer.php (hunk 2 above)
Comment (michael-e) [reply install/lib/class.installer.php]: Thanks for keeping an eye on this!

File: install/lib/class.updater.php (hunk 3 above)
Comment (nitriques) [reply install/lib/class.updater.php]: No this check it there to make sure you are running at least 2.7.0 when updating to 3.0.0.

File: install/lib/class.installer.php (hunk 2 above)
Comment (nitriques) [reply install/lib/class.installer.php]: @michael-e np!

**Comment by nitriques** [comment]
@fhamon check 58876a1

File: symphony/lib/toolkit/class.extensionmanager.php [hunk 4]
```diff
@@ -1027,27 +1029,15 @@ public static function about($name, $rawXML = false)
...
                 // Min version
-                if (!empty($required_min_version) && version_compare($current_symphony_version, $required_min_version, '<')) {
+                if (!empty($required_min_version) && \Composer\Semver\Comparator::lessThan($current_symphony_version, $required_min_version)) {
                     $about['status'][] = Extension::EXTENSION_NOT_COMPATIBLE;
                     $about['required_version'] = $required_min_version;
 
-                    // Max version
-                } elseif (!empty($required_max_version) && version_compare($current_symphony_version, $required_max_version, '>')) {
+                // Max version
+                } elseif (!empty($required_max_version) && \Composer\Semver\Comparator::lessThan($current_symphony_version, $required_max_version)) {
```
Comment (brendo) [inline symphony/lib/toolkit/class.extensionmanager.php]: I believe this one should be greater than?

File: install/lib/class.updater.php (hunk 3 above)
Comment (fhamon) [reply install/lib/class.updater.php]: :+1:

File: symphony/lib/toolkit/class.extensionmanager.php (hunk 4 above)
Comment (nitriques) [reply symphony/lib/toolkit/class.extensionmanager.php]: @brendo Fixed in 0c95af85a2. Can I get a green check mark now ? haha! Thanks buddy.

**Pull request merged**